### PR TITLE
AP2 layout with media keys

### PR DESCRIPTION
**Pull request by ghost**
## Description
New keymap adds tap AP2 feature, media keys and remap backlight control as in AP2.
Contains very detailed txt manual.

## Types of Changes
- [ ] Core
- [x] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/anne_pro/anne_pro.c [hunk 1]
```diff
@@ -173,7 +173,7 @@ void keyboard_post_init_kb(void) {
     /* Set the theme to rainbow */
     anne_pro_lighting_mode(APL_MODE_RAINBOW);
     /* Set the effect rate to average and the brightness to average */
-    anne_pro_lighting_rate_brightness(128, 5);
+    anne_pro_lighting_rate_brightness(128, 10);
```
Comment (msvisser) [inline keyboards/anne_pro/anne_pro.c]: This might be your personal preference, however this change is not really related to the layout, so it should probably not be included in this pull request. I will probably make these things config options at some point.

File: keyboards/anne_pro/keymaps/ap2style_win/keymap.c [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  [1] = LAYOUT(
+      KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  XXXXXXX,
+      XXXXXXX, XXXXXXX, KC_UP,   XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, KC_UP,   KC_SLCK, KC_PAUS, KC_HOME, KC_END,  KC_PSCR,
+      XXXXXXX, KC_LEFT, KC_DOWN, KC_RGHT, XXXXXXX, XXXXXXX, XXXXXXX, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGUP, KC_PGDN,          XXXXXXX,
+      _______, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, TG(3),   XXXXXXX, XXXXXXX, XXXXXXX, KC_INS,  KC_DEL,                   _______,
+      _______, _______, _______,                   XXXXXXX,                                     _______, _______, _______, _______
+  ),
+  [2] = LAYOUT(
+      XXXXXXX,   KC_MSTP, KC_VOLD, KC_VOLU, KC_MPRV, KC_MPLY, KC_MNXT, KC_MUTE, XXXXXXX, APL_MOD, APL_RGB, APL_RAT, APL_BRT, XXXXXXX,
+      RESET, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
```
Comment (msvisser) [inline keyboards/anne_pro/keymaps/ap2style_win/keymap.c]: Why did you move RESET to the TAB key instead of ESC? This is not consistent with the rest of the layouts.

File: keyboards/anne_pro/keymaps/ap2style_win/keymap.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#include QMK_KEYBOARD_H
+
+/* Default Anne Pro layout, arrows style */
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
+      KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+      KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,                     MT(MOD_RSFT, KC_UP),
+      KC_LCTL, KC_LWIN,   KC_LALT,                   KC_SPACE,                                    KC_LALT, LT(1,KC_LEFT), LT(2,KC_DOWN), MT(MOD_RCTL, KC_RGHT)
```
Comment (msvisser) [inline keyboards/anne_pro/keymaps/ap2style_win/keymap.c]: Alignment of the keycodes is off, here and in some other places in the keymap.

File: keyboards/anne_pro/keymaps/ap2style_win/keymap.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+/* Default Anne Pro layout, arrows style */
```
Comment (msvisser) [inline keyboards/anne_pro/keymaps/ap2style_win/keymap.c]: Maybe change this comment to better reflect the actual type of layout.

**Review by msvisser — CHANGES_REQUESTED** [review]
First of all, thank you very much for wanting to contribute!

There are some small comments I have made which I would like to see resolved before merging the layout. However this layout is a nice addition to the collection of defaults.

Furthermore I have some suggestions

- It might be nice to move the Anne Pro 2 style layouts into a subfolder. I am not sure if QMK supports this, but it would be awesome if you could check it out!
- If you want to it might be nice to also have a macOS style layout with the same features

Overall it looks good! Again, thank you very much for contributing!

File: keyboards/anne_pro/keymaps/ap2style_win/keymap.c (hunk 3 above)
Comment (ghost) [reply keyboards/anne_pro/keymaps/ap2style_win/keymap.c]: I dont know how and where to "on" keymap alignment

**Comment by angelokofficial** [comment]
I'm making some changes now. Should be better soon.

**Comment by ghost** [comment]
I tried all my best to find and make a method to build firmware with a keymap from keymap subfolder. I found something like this only for [layouts](https://docs.qmk.fm/#/feature_layouts?id=layouts-using-a-keymap-with-multiple-keyboards) (which relate to the repository, not to the kb folder), but that's probably not what we need. I didn't found the concrete method from QMK side to do this with keymaps.

However, I tried to put in order my layouts to make them look better. And I fixed some issues which I did for my preferences. I'm really sorry I can't do easy work properly.

**Comment by msvisser** [comment]
Looks good! Thank you for contributing to this project!
